databricks/koalas/frame.py

```diff
@@ -2690,19 +2715,45 @@ def fillna(self, value=None, axis=None, inplace=False):
             axis = 0
         if not (axis == 0 or axis == "index"):
             raise NotImplementedError("fillna currently only works for axis=0 or axis='index'")
+        if (value is None) and (method is None):
+            raise ValueError("Must specify a fill 'value' or 'method'.")
 
-        if value is None:
-            raise ValueError('Currently must specify value')
-        if not isinstance(value, (float, int, str, bool, dict, pd.Series)):
-            raise TypeError("Unsupported type %s" % type(value))
-        if isinstance(value, pd.Series):
-            value = value.to_dict()
-        if isinstance(value, dict):
-            for v in value.values():
-                if not isinstance(v, (float, int, str, bool)):
-                    raise TypeError("Unsupported type %s" % type(v))
-
-        sdf = self._sdf.fillna(value)
+        sdf = self._sdf
+        if value is not None:
+            if not isinstance(value, (float, int, str, bool, dict, pd.Series)):
+                raise TypeError("Unsupported type %s" % type(value))
+            if isinstance(value, pd.Series):
+                value = value.to_dict()
+            if isinstance(value, dict):
+                for v in value.values():
+                    if not isinstance(v, (float, int, str, bool)):
+                        raise TypeError("Unsupported type %s" % type(v))
+            if limit is not None:
+                raise ValueError('limit parameter for value is not support now')
+            sdf = sdf.fillna(value)
+        else:
+            for data_column in self._internal.data_columns:
+                if method in ['ffill', 'pad']:
+                    func = F.last
+                    end = (Window.currentRow - 1)
+                    if limit is not None:
+                        begin = Window.currentRow - limit
+                    else:
+                        begin = Window.unboundedPreceding
+                elif method in ['bfill', 'backfill']:
+                    func = F.first
+                    begin = Window.currentRow + 1
+                    if limit is not None:
+                        end = Window.currentRow + limit
+                    else:
+                        end = Window.unboundedFollowing
+                else:
+                    raise ValueError('Expecting pad, ffill, backfill or bfill.')
+                window = Window.orderBy(self._internal.index_columns[0]).rowsBetween(begin, end)
```

Comment:
Can we support multi-index or no index?
If we can, let's support while we're here, otherwise let's check the index size?

Reply:
Yea, I think supporting multi index isn't that difficult. @RainFung, can you try to fix `self._internal.index_columns[0]` -> `self._internal.index_columns` and add a test with multi index? If that doesn't work, I will make a followup to fix.

Reply:
fixed